### DISCUSSION TITLE
Which writer by default wire api uses while encoding protobuf message

### DISCUSSION THREAD

**Discussion by kspmmitt** [square/wire#3063] [General]
Hello,
Can you please tell which Writer (out of ProtoWriter and ReverseProtoWriter) is used while encoding protobuf message? Also plz tell situations where both of writers are used?
Asking above w.r.t issue# https://github.com/square/wire/discussions/2755 where it says ProtoWriter is more performant?

**Comment by oldergod** [comment]
`ReverseProtoWriter` is the default and should be the most efficient. We have not had the time to dig into #2755
